### PR TITLE
toix URL validator

### PR DESCRIPTION
**Pull request by ghost**
URL validation functionality to ```toix```. The function tries to convert the given string to URL or fail if the given string is not a valid URL. The [URL constructor](https://developer.mozilla.org/en-US/docs/Web/API/URL) creates and returns an URL object composed from the given parameters. Optional parameters are port and protocol.

Version bump version to ```1.0.6```.

### CODE REVIEW

**Comment by ghost** [comment]
I figured out we could use https://www.npmjs.com/package/is-url instead of going directly with RegExp.

There is only one thing I encountered that might be a problem in the future, that is, the module allows ```"http://www.google"``` type of URL when it shouldn't. 

```
      is_url()
        ✓ should allow null
        ✓ should allow undefined
        ✓ should allow "http://www.google.com"
        ✓ should allow "https://www.google.com"
        ✓ should allow "http://google.com"
        ✓ should allow "http://www.google.com/imghp"
        ✓ should allow "http://g.co"
        ✓ should not allow "www.google"
        1) should not allow "http://www.google"
        ✓ should not allow "http://google"
        ✓ should not allow "www.google#.com"
        ✓ should not allow "www.google-.co"
        ✓ should not allow "www.-google.co"


  116 passing (34ms)
  1 failing

  1) toix
       str
         is_url()
           should not allow "http://www.google":
     Error: Allowed!
      at Context.it (index.spec.ts:76:15)
```

Edit 1: I opened an issue @ https://github.com/segmentio/is-url/issues/27

File: packages/toix/package.json [hunk 1]
```diff
@@ -3,7 +3,11 @@
   "version": "1.0.5",
```
Comment (hf) [inline packages/toix/package.json]: Bump version please.

Comment (ghost) [reply packages/toix/package.json]: 1.0.6?

File: packages/toix/index.ts [hunk 2]
```diff
@@ -25,6 +25,38 @@ export namespace str {
...
+  export const url = <X extends string, R extends string, O extends string>(options: { protocol?: R, port?: O  } = {}) =>
+    wrap(
+      "str.url",
+      transform<X, URL & { protocol: R, port: O }>(
+        value => {
+          try {
+            const url = new URL(value);
+
+            if (options && options.port && options.port !== url.port) {
+              throw new ValidationError(`not a ${options.port} port`, value);
```
Comment (hf) [inline packages/toix/index.ts]: Maybe reformulate this error message. It sounds a bit weird.

File: packages/toix/package.json (hunk 1 above)
Comment (hf) [reply packages/toix/package.json]: Yup

File: packages/toix/package.json [hunk 3]
```diff
@@ -13,18 +17,27 @@
...
+    "schema",
+    "validation"
+  ],
   "author": "Stojan Dimitrovski <sdimitrovski@gmail.com>",
   "license": "MIT",
   "dependencies": {
     "@toi/toi": "^1.0.0",
-    "isemail": "^3.1.3"
+    "isemail": "^3.1.3",
+    "node": "^11.0.0"
```
Comment (hf) [inline packages/toix/package.json]: Why not `node@^10.0.0`?

File: packages/toix/package.json [hunk 4]
```diff
@@ -13,18 +17,27 @@
...
+    "schema",
+    "validation"
+  ],
   "author": "Stojan Dimitrovski <sdimitrovski@gmail.com>",
   "license": "MIT",
   "dependencies": {
     "@toi/toi": "^1.0.0",
-    "isemail": "^3.1.3"
+    "isemail": "^3.1.3",
+    "node": "^10.0.0"
```
Comment (hfsd) [inline packages/toix/package.json]: Actually, maybe do ">= 10.0.0"

**Review by hf — APPROVED** [review]
Works OK, really useful. Thanks!

**Pull request merged**